### DISCUSSION TITLE
How is HEAT_OF_VAPORIZATION calculated?

### DISCUSSION THREAD

**Discussion by abc-k** [firemodels/fds#12216] [Q&A] [answered]
I am trying to burn plastics (ex. polyethylene) with HRR fixed. This is to observe how the vaporized fuel ignites. To do this, I created a sample file and varied some parameters. Then I found something I don't understand about HEAT_OF_VAPORIZATION.

In the sample file posted in the reply, polyethylene was set as the solid fuel, referring to User Guide 11.4.4.

_“FDS_User_Guide 11.4.4_
_&MATL ID = 'stuff'_
_CONDUCTIVITY = 0.1_ 
_SPECIFIC_HEAT = 1.0_ 
_DENSITY = 900.0 /_

_&SURF ID = 'my surface'_ 
_COLOR = 'GREEN'_
_MATL_ID = 'stuff'_ 
_HRRPUA = 1000._ 
_IGNITION_TEMPERATURE = 500._ 
_HEAT_OF_VAPORIZATION = 1000._
_RAMP_Q = 'fire_ramp'_ 
_THICKNESS = 0.01 /_ 

_&RAMP ID='fire_ramp', T= 0.0, F=0.0 /_
_&RAMP ID='fire_ramp', T= 10.0, F=1.0 /_ 
_&RAMP ID='fire_ramp', T=310.0, F=1.0 /_ 
_&RAMP ID='fire_ramp', T=320.0, F=0.0 /“_

The changes from FDS_User_Guide are shown below.
(1) Warm the ignition source to 2500°C to reach the ignition temperature.
(2) Lower the ignition temperature to 100°C to facilitate ignition.
(3) Set polyethylene as fuel.

Under these conditions, the value of HEAT_OF_VAPORIZATION was changed to 100, 1000, and 3000. The output shows that the HRR does not change, but the ignition source temperature changes.
![image](https://github.com/firemodels/fds/assets/150226734/cf294657-a243-4376-9992-cf28925db7a6)

![image](https://github.com/firemodels/fds/assets/150226734/bc7a0a70-ab22-498f-b7b9-c25a8e5f1aa0)

However, I could not understand how this was calculated and output, and what effect this value has.

Could you tell me how the value of HEAT_OF_VAPORIZATION is used in the calculation inside the FDS and what is the effect of this value?

**Comment by abc-k** [comment]
! This sample file is HEAT_OF_VAPORIZATION=100.
! I changed only this parameter to 1000 and 3000.

&HEAD CHID='simple_testVA100', TITLE='Simple demonstration case.'  / 

&MESH IJK=36,24,24, XB=0.0,3.6,0.0,2.4,0.0,2.4 / 

&TIME T_END=320.0 /  

&REAC FUEL='POLYETHYLENE', SOOT_YIELD=0.01,FORMULA='C2H4',HEAT_OF_COMBUSTION=40000 /  

&MATL ID = 'stuff' 
CONDUCTIVITY = 0.1 
SPECIFIC_HEAT = 1.0 
DENSITY = 900.0 / 

&SURF ID = 'BURNER' 
COLOR = 'GREEN' 
MATL_ID = 'stuff' 
HRRPUA = 1000. 
IGNITION_TEMPERATURE = 100. 
HEAT_OF_VAPORIZATION = 100.
RAMP_Q = 'fire_ramp' 
THICKNESS = 0.01 / 

&RAMP ID='fire_ramp', T= 0.0, F=0.0 /
&RAMP ID='fire_ramp', T= 10.0, F=1.0 / 
&RAMP ID='fire_ramp', T=310.0, F=1.0 / 
&RAMP ID='fire_ramp', T=320.0, F=0.0 /

!&SURF ID='BURNER', HRRPUA=1000., COLOR='RED' /

&SURF ID='HOT', TMP_FRONT=2500., COLOR='BLACK' /



&OBST XB=0.0,0.4,1.0,1.4,0.0,0.2, SURF_ID='HOT'/
&VENT XB=0.0,0.4,1.0,1.4,0.2,0.2, SURF_ID='BURNER' /

&VENT XB=3.6,3.6,0.8,1.6,0.0,2.0, SURF_ID='OPEN' / 

&BNDF QUANTITY='GAUGE HEAT FLUX' / 

&DEVC ID='FIRE_SURFACE_TEMP' XB=0.0,0.4,1.0,1.4,0.0,0.2,QUANTITY='TEMPERATURE'/
&DEVC ID='OBJECT_SURFACE_TEMP',XB = 0.0, 0.4, 1.0, 1.4, 0.2, 0.2, QUANTITY = 'WALL TEMPERATURE', IOR = 3 /


&SLCF PBY=1.2, QUANTITY='TEMPERATURE', VECTOR=.TRUE. /

&TAIL /

**Comment by drjfloyd** [comment]
You are specifying the heat release rate with HRRPUA.  That value is fixed and HEAT_OF_VAPORIZATION has no impact on it. HEAT_OF_VAPORIZATION is used in the solid phase 1D temperature calculation. Energy is removed from the solid at a rate of HRRPUA * HEAT_OF_VAPORIZATION, this helps keep the surface temperature from rising to unreasonably high values.

**Reply by abc-k** [reply]
Appreciate the reply. I have a few questions that I don't understand.
(1) When you say _"solid phase 1D temperature calculation"_, do you mean the _"1D solid phase solver"_ mentioned in the user guide?
(2) By _"energy"_ do you mean flame energy?
(3) What is the source of the HRRPUA*HEAT OF VAPORIZATION formula?
Could you help me with the above three points?

**Comment by drjfloyd** [comment]
(1) Yes
(2) No.  See Eq 7.2 in the Technical Refernce Guide. HEAT_OF_VAPORIZATION is used to define q'''_s
(3) I was a little too quick in that response. We want to get q'''_s, the energy that it takes to convert the solid to the gas that can then burn in the gase phase.  The units of q'''_s is kW/m3. The units of HRRPUA is kW/m2 which is actually a mass loss rate (HRRPUA / HEAT_OF_COMBUTSTION = kg/m2/s of fuel from the solid).  The units of the HEAT_OF_VAPORIZATION is kJ/kg.  So if we want to get how much energy the solid phase must supply to vaporize the fuel needed to give the HRRPUA: 
HRRPUA / HEAT_OF_COMBUSTION * HEAT_OF_VAPORIZATION = the kW/m2 the solid phase supplies to vaporize the fuel. This energy is removed from the first wall node so q'''_s is obtained by dividing the kW/m2 by the node thickness (kW/m2 / m = kW/m3).

**Reply by abc-k** [reply]
Thank you for teaching. 
I think I can understand the calculation of HEAT_OF_VAPORIZATION.

**Reply by abc-k** [reply]
@drjfloyd 　Dear,drjfloyd.
Could you tell me about the terms used in deriving Q_S(1)?

From the source code, I found the following equation

_Wall.f90(source)_
   _Q_S(1) = Q_S(1) - B1%M_DOT_G_PP_ADJUST(REACTION(1)%FUEL_SMIX_INDEX)*SF%H_V/DX_S(1)_

From the formula form,
_-B1%M_DOT_G_PPP_ADJUST(REACTION(1)%FUEL_SMIX_INDEX)*SF%H_V/DX_S(1)_
is **the q'''_s, the energy that it takes to convert the solid to the gas**, as we were told by you. We also know that _SF%H_V_ in this equation is the _HEAT_OF_VAPORIZATION_ described in the SURF line. However, I do not understand what the other terms (_B1%M_DOT_G_PP_ADJUST(REACTION(1)%FUEL_SMIX_INDEX)_ and _DX_S(1)_) represent.

Could you please tell me what these two variables represent?

**Comment by drjfloyd** [comment]
If you are going to be working with the source code you should try to get familiar with tracing variables through the code to see how they are used.  It will take a little time at first, but once you get some experience it will get faster. In many cases we do try to give  a name that hints at what the variable does or in some cases point to a specific equation in the Tech Ref. M_DOT_G_PP_ADJUST is defined in the previous block of code where the comment is: Compute the mass flux of reaction gases at the surface.  M for mass, DOT for rate of change (the dot used above symbols in equations in the Tech Ref), G for gas, PP for double prime ('' or per area).  DX_S(1).  DX, DY, DZ, DT, etc. these all indicate a discretized variable. For DX_S we have X for distances, S for solid; and (1) indicates it is the first item in an array. Earlier in the routine, DX_S is tied to either a surface defintion when there is no pyrolysis or to the wall arrays for a specific wall cell or particle when there is pyrolysis. DX_S is the thickness of the first wall node.

**Reply by abc-k** [reply]
Thank you for teaching.
I now understand that M_DOT_G_PP_ADJUST refers to the formula (7.36) in Tech Ref.
And I also understand how _"HRRPUA / HEAT_OF_COMBUSTION * HEAT_OF_VAPORIZATION"_, as I was told by you last time, is handled in the FDS.

**Comment by abc-k** [comment]
What is the relationship between q'''_s (source code value) and q'''_s (Tech Ref. Eq. 7.2, second term on the right side)?

I was previously told that HEAT_OF_VAPORIZATION is used in the definition of q'''_s (_the energy that it takes to convert the solid to the gas that can then burn in the gaseous phase_).
 I also found from the source code that q'''_s (=HRRPUA*HEAT_OF_VAPORIZATION/HEAT_OF_COMBUSTION/node thickness) is subtracted from Q_S(1).

However, q'''_s (Tech Ref. Eq. 7.2) derives q'''_s in a different way than the source code equation. (Tech Ref. Eq. 7.3)

Could you please explain the difference between these two variables (q'''_s (=HRRPUA*HEAT OF VAPORIZATION/HEAT OF COMBUSTION/node thickness) and q'''_s (Tech Ref. Eq. 7.3))?

**Comment by drjfloyd** [comment]
7.2 is the conservation of energy for a solid element in derivative form.  From left to right the terms are heat storage in the solid due to its temperature, conduction of heat through the solid, and heat generation in the solid.

7.3 is simply noting that there are multiple ways to generate heat in the solid. There are chemical processes (chemical reactions [e.g. pyrolysis], evaporation, condenstation, etc.) and there is the possibility for radiation to be absorbed in-depth. 

In your input file you do not have radiation absorption in-depth so the q dot s,r ''' term is 0 leaving just the chemical term which you do have in the form of the HEAT_OF_VAPORIZATION input.

**Reply by abc-k** [reply]
Thank you for your teaching.

I understand that it is derived that q_dot_s'''= q_dot_s,c''' (chemical term in Tech Ref . Eq. 7.3)　in this input file.
And I found that this q_dot_s,c''' is derived by Tech Ref. Eq. 7.46.
However, I do not understand which term in q_dot_s,c'''(Tech Ref. Eq. 7.46) is assigned to HEAT_OF_VAPORIZATION.
Thus, I would like to ask the following three questions.

(1) Could you please tell me which term in Tech Ref. Eq. 7.46 is assigned to HEAT_OF_VAPORIZATION?
(2) Could this equation for q_dot_s,c''' be transformed into the form q'''_s?
(3) What kind of device can be added to output the effect of this HEAT_OF_VAPORIZATION?
　I tried adding HRRPUV REAC in 3 cases with HEAT_OF_VAPORIZATIO =100, 1000, and 3000. However, in all cases, 0 was output at all times and I could not see the effect. The HOV in the figure below indicates HEAT_OF_VAPORIZATION.

![image](https://github.com/firemodels/fds/assets/150226734/9280e92e-156f-4c85-b1ea-ffc51884d00e)

**Comment by drjfloyd** [comment]
1) Eq 7.46 is the mass transfer coefficient. The HEAT_OF_VAPORIZATION does not appear in that equation
2) q_dot_s,c is q_dot_s in your case
3) HRRPUV REAC is for gas phase and not solid phase reactions.  HEAT_OF_VAPORIZATION is for the solid phase.  All of the solid phase output quantities are in the table of solid phase outputs in the User's Guide.

**Comment by drjfloyd** [comment]
I see I think you mean Eq. 7.45.  This equation says that the volumetric heat source (kW/m3) in the solid is the sum over all solid phase reactions of the reaction rate in kg/m3/s times a heat of reaction in kJ/kg.  Which of these does the HEAT_OF_VAPORIZATION correspond to?

**Reply by abc-k** [reply]
I apologize. I mis-numbered the equation. Eq. 7.45 is correct, as you were told by me.
I think HEAT_OF_VAPORIZATION is not in any of the terms of Eq. 7.45. Hence, I am not sure why A and B shown below are equal.

A: q'''_s (_=HRRPUA*HEAT_OF_VAPORIZATION/HEAT_OF_COMBUSTION/node thickness=-B1%M_DOT_G_PPP_ADJUST(REACTION(1)%FUEL_SMIX_INDEX)*SF%H_V/ DX_S(1)_)
A is the variable used in the source code.

B: q'''_s (Tech Ref. Eq. 7.2) and now,in my case, q'''_s= q_dot_s,c'''
B is a variable with the same name as A, as described in the Tech Ref..

I looked for where q'''_s of A is listed in Tech Ref. and found q'''_s of B. I think they are the same thing. However, there is a HEAT_OF_VAPORIZATION in A, whereas there is no HEAT_OF_VAPORIZATION in B.


My input file assumed solid fuel, but is it actually calculated as gaseous or liquid fuel in FDS?

In my input file, HEAT_OF_VAPORIZATION is assigned as heat of reaction and therefore not found in the B equation?

**Comment by drjfloyd** [comment]
The HEAT_OF_VAPORIZATION is used as the heat of reaction. Your fuel is still a solid. The liquid evaporation model isn't invoked unless you have a boiling temperature.

**Reply by abc-k** [reply]
Thank you for teaching .
I now understand how HEAT_OF_VAPORIZATION is used.

Also, thank you for telling me what QUANTITY I can use in the solid phase output.
I have referred to the technical reference guide and the user guide. And I thought that q_dot_s,c'''' is CONVECTIVE HEAT FLUX.


I added CONVECTIVE HEAT FLUX to each of my files （HEAT_OF_VAPORIZATION=100,1000,3000） and output. However, as shown in the figure below, the output results did not change at all.
![image](https://github.com/firemodels/fds/assets/150226734/7bcc0805-8d63-441a-88d0-5a0c2a07612f)

Also, the Q_CONV results output in HRR.csv output exactly the same values in all cases. I also tried RADIATIVE HEAT FLUX and TOTAL HEAT FLUX just to be sure, but they all produced the same results.
![image](https://github.com/firemodels/fds/assets/150226734/92a20bbd-429b-4c7b-9e98-2b9c3d72a13e)

Besides these two parameters (Q_CONV and CONVECTIVE HEAT FLUX), I could not find any other parameters that could output q'''_s (or q_dot_s,c''').

Could you please help me with the following two questions
(1) Is CONVECTIVE HEAT FLUX a parameter that outputs q_dot_s,c'''?
(2) Which QUANTITY could be used to output the effect of changing HEAT_OF_VAPORIZATION?

**Comment by drjfloyd** [comment, marked as answer]
Q_CONV in the hrr.csv file is not a solid phase quantity. See Section 22.10.1 of the User's Guide for how this term is defined.
CONVECTIVE HEAT FLUX is the heat flux into (or out of) the surface due to convection. It is not a measure of enthalpy inside the surface.

There isn't an output for q_dot_s'':
-Since you have a fixed HRRPUA with a HEAT_OF_VAPORIZATION, you can compute the value of this term which exists on in the first wall node in your case as you know the burning rate, H_O_V, and the first wall node thickness (look at the .out file).
-If you want to verify this you can add outputs to compute the energy balance in the first wall node. The net heat flux at the surface minus conduction heat transfer out of the first wall node will equal the enthalpy change of the first wall node due to its temperature change plus enthalpy change due to the H_O_V. You know the CONDUCTIVTY, DENSITY, SPECIFIC_HEAT, and wall node spacing. You can output the INSIDE WALL TEMPERATURE and NET HEAT FLUX. You have everything you need to compute the energy balance in the first wall node.

That being said if I take your input file from your second post and add the line:
&DEVC XYZ=0.2,1.2,0.2,IOR=3,QUANTITY='CONVECTIVE HEAT FLUX'/

and run with H_O_V = 100 (blue) and 3000 (orange):
![image](https://github.com/firemodels/fds/assets/12799217/3512bbf8-9c4e-4ec0-9d8e-cb93f7aacbe8)

100 will remove less heat from the surface: the surface temperature will be higher than 3000, higher surface temperature means less temperature difference from the gas, and less temperature difference means less convective flux.

**Reply by abc-k** [reply]
Thank you for teaching.
I found the FIRST WALL NODE THICKNESS from the .out file. ( Look at the highlights in the figure shown below).
![image](https://github.com/firemodels/fds/assets/150226734/06ee3898-27b0-464d-9a9f-6965ae614dfe)

I thought _“The net heat flux at the surface minus conduction heat transfer out of the first wall node will equal the enthalpy change of the first wall node due to its temperature change plus enthalpy change due to the H_O_V.”_ as you taught me, as in the following equation.  ' name ' indicates that the fds output or input value is being used.
![image](https://github.com/firemodels/fds/assets/150226734/3c06a2a0-25ea-4062-99fe-1994bac34e7e)

The equation of applying this to each case are shown in the figure below. 
![image](https://github.com/firemodels/fds/assets/150226734/9df8f773-848b-4272-acf0-43e35a02a17e)

 I think I have obtained q_dot_s".
Could you please tell me if my understanding of how to obtain "q_dot_s" is correct?

**Comment by drjfloyd** [comment]
Until the material reaches its ignition temperature, the HRRPUA is zero and hence the heat of reaction source term is zero. Does your plot show this?

Pleaes carefuly review the FDS technical reference guide and/or an introductory text on heat transfer on how to model 1D heat transfer.

**Reply by abc-k** [reply]
Thank you very much.

I was able to derive Q_DOT_S''' by the approach you taught me before.
It took about 136[s] to reach the ignition temperature.
As shown in the figure below, I could confirm that each Q_DOT_S''' changes according to the value of HEAT_OF_VAPORIZATION after reaching IGNITION_TEMPERATURE.  
![image](https://github.com/firemodels/fds/assets/150226734/d124cf45-25d8-4e78-af03-73345c5f9119)

I will make sure to read carefully the TECH REF. and introductory text on heat transfer on how to model 1D heat transfer.

Thank you for your time.
